extensions/resource-deployment/src/ui/resourceTypeWizard.ts

```diff
@@ -85,27 +83,57 @@ export class ResourceTypeWizard {
 		}));
 
 		this.toDispose.push(this.wizardObject.doneButton.onClick(async () => {
-			await this.model.onOk();
+			await this._model.onOk();
 			this.dispose();
 		}));
 		this.toDispose.push(this.wizardObject.generateScriptButton.onClick(async () => {
-			await this.model.onGenerateScript();
+			await this._model.onGenerateScript();
 			this.dispose();
 			this.wizardObject.close(); // close the wizard. This is already hooked up into doneButton, so it is not needed for that button above.
 		}));
 		this.toDispose.push(this.wizardObject.cancelButton.onClick(() => {
-			this.model.onCancel();
+			this._model.onCancel();
 			this.dispose();
 		}));
+	}
 
+
+	private getResourceProviderModel(): ResourceTypeModel {
+		if (instanceOfWizardDeploymentProvider(this.provider)) {
+			return new DeployClusterWizardModel(this.provider, this);
+		} else if (instanceOfAzureSQLVMDeploymentProvider(this.provider)) {
+			return new DeployAzureSQLVMWizardModel(this.provider, this);
+		} else if (instanceOfNotebookWizardDeploymentProvider(this.provider)) {
+			return new NotebookWizardModel(this.provider, this);
+		} else if (instanceOfAzureSQLDBDeploymentProvider(this.provider)) {
+			return new DeployAzureSQLDBWizardModel(this.provider, this);
+		} else {
+			return new PageLessDeploymentModel(this.provider, this);
+		}
+	}
+
+	public async open(): Promise<void> {
+		/**
+		 * Currently changing wizard titles and pages does not work without closing and reopening the wizard. (it makes the changes to objects but visually everything remains the same).
+		 * Also, the done button listener gets broken when we close and reopen the same dialog
+		 * For these reasons, I am creating a new wizard every time user changes the options.
```

Comment:
... user changes the options that requires changes to the wizard's titles and pages.